src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,190 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
```

Comment:
```suggestion
    private Node<T> tail;
    private int size;
```
lets store current size, it will make your code faster
no need to count elements each time you need the value